### PR TITLE
fix(OSD-24257): put clusters in LS if DMS/PD is blocked

### PR DESCRIPTION
**Pull request by typeid**
This PR updates CAD's CHGM behavior to put clusters that block PD/DMS into limited support. We need to do this as we can't guarantee SLA in those cases.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ run:
   modules-download-mode: readonly
   # Timeout for analysis, e.g. 30s, 5m.
   # Default: 1m
-  timeout: 5m
+  timeout: 15m
```
Comment (typeid) [inline .golangci.yml:6]: With the new binaries we have in this repo, we often time out with 5 min.

File: cadctl/cmd/investigate/investigate.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ func run(_ *cobra.Command, _ []string) error {
 		return ccam.Evaluate(cluster, err, ocmClient, pdClient, alertInvestigation.Name)
 	}
 
-	investigationResources := &investigation.Resources{InvestigationName: alertInvestigation.Name, Cluster: cluster, ClusterDeployment: clusterDeployment, AwsClient: customerAwsClient, OcmClient: ocmClient, PdClient: pdClient}
+	investigationResources := &investigation.Resources{Cluster: cluster, ClusterDeployment: clusterDeployment, AwsClient: customerAwsClient, OcmClient: ocmClient, PdClient: pdClient}
```
Comment (typeid) [inline cadctl/cmd/investigate/investigate.go:126]: We didn't need the investigation name, it's better to have it as variable in the investigations to reduce bloat here.

File: pkg/investigations/chgm/chgm.go [hunk 3]
```diff
@@ -22,21 +22,17 @@ import (
...
-	InternalOnly: false,
-}
+	chgmSL = ocm.ServiceLog{
+		Severity:     "Critical",
+		Summary:      "Action required: cluster not checking in",
+		ServiceName:  "SREManualAction",
+		Description:  "Your cluster is no longer checking in with Red Hat OpenShift Cluster Manager. Possible causes include stopped instances or a networking misconfiguration. If you have stopped the cluster instances, please start them again - stopping instances is not supported. If you intended to terminate this cluster then please delete the cluster in the Red Hat console",
+		InternalOnly: false,
+	}
+)
```
Comment (typeid) [inline pkg/investigations/chgm/chgm.go:35]: These were global variables we modify, which could lead to odd behavior. I created helper functions to create them.

File: pkg/investigations/chgm/chgm.go [hunk 4]
```diff
@@ -110,6 +98,37 @@ func Investigate(r *investigation.Resources) error {
...
+			break
+		}
+	}
+
+	if setLS {
+		err := ocmCli.PostLimitedSupportReason(createEgressLS(egressFailures), cluster.ID())
+		if err != nil {
+			return err
+		}
+		metrics.Inc(metrics.LimitedSupportSet, investigationName, "EgressBlocked")
```
Comment (typeid) [inline pkg/investigations/chgm/chgm.go:120]: thought: we often run into issues with `metrics.Inc` due to the nature of it accepting everything, we should wrap the metric increases into functions to properly check parameters.

Comment (RaphaelBut) [reply pkg/investigations/chgm/chgm.go:120]: agreed

**Pull request merged**